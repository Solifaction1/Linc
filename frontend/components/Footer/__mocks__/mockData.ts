/* eslint-disable max-lines */
import { Contact } from "@models/contact.model"
import { Navigation, RelatedContactInformation } from "@models/navigation.model"

const mockNavigationParent1: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 1,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'parent1',
    related: null,
    sitemap_exclude: true,
    title: 'Parent 1',
    type: '',
    uiRouterKey: 'parent1',
    updatedAt: ''
}

const mockNavigationParent2: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 4,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'parent2',
    related: null,
    sitemap_exclude: true,
    title: 'Parent 2',
    type: '',
    uiRouterKey: 'parent2',
    updatedAt: ''
}


export const mockContact: Contact = {
    id: 10,
    street: 'street',
    houseNumber: 'houseNumber',
    postalCode: 0,
    city: 'city',
    telephone: '+1234567890'
}

const mockContactInformationRelated: RelatedContactInformation = {
    createdAt: 'createdAt',
    facebook: 'facebook',
    id: 7,
    instagram: 'instagram',
    linkedin: 'linkedin',
    locale: 'en',
    navigationItemId: 0,
    publishedAt: 'publishedAt',
    sitemap_exclude: false,
    updatedAt: 'updatedAt',
    updatedBy: 'updatedBy',
    __contentType: '__contentType'
}

const mockContactInformation: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 8,
    items: null,
    menuAttached: true,
    order: 0,
    parent: null,
    path: 'contact-information',
    related: mockContactInformationRelated,
    sitemap_exclude: true,
    title: 'Contact Information',
    type: '',
    uiRouterKey: 'contact-information',
    updatedAt: ''
}

const mockNavigationDataCompliantLinks: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 2,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'data-compliant-links',
    related: null,
    sitemap_exclude: true,
    title: 'Data Compliant Links',
    type: '',
    uiRouterKey: 'data-compliant-links',
    updatedAt: ''
}

export const mockNavigationData = [
    mockNavigationParent1,
    mockNavigationParent2,
    mockContactInformation,
    mockNavigationDataCompliantLinks,
]