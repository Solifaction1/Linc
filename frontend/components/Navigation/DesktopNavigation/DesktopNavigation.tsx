import React from 'react'
import Image from 'next/image'
import Link from 'next/link'
import { Navigation } from '@models/navigation.model'
import LanguageSwitch from '@components/Language-switch/LanguageSwitch'
import { Localizations } from '@models/attributes.model'
import SwitchCTA from '@templates/SwitchCTA'
import { getColorHeader } from '@constants/color'
import EmployeeData from '@models/employee.model'
import EmployeeSwitch from '@components/EmployeeSwitch/EmployeeSwitch'
import { getUniqueId } from '@utils/utils.service'
import styles from './desktopNavigation.module.scss'

const DesktopNavigation = ({
	internalLinksData,
	employeesData,
	className,
	contactButtonClassName,
	localizations,
	color,
}: {
	internalLinksData: Navigation[]
	employeesData: { data: EmployeeData[] }
	className: string
	contactButtonClassName: string
	localizations: Localizations | undefined
	color: string | undefined
}) => (
	<div className={`${styles.navigationWrapper} ${className}`}>
		<Link href="/">
			{color === 'beige' || color === 'blue-gray' || color === 'white' ? (
				<Image
					src="/images/logo-gray.svg"
					alt="logo"
					width={172}
					height={68}
					className={styles.logo}
					priority
					quality={60}
				/>
			) : (
				<Image
					src="/images/logo.svg"
					alt="logo"
					width={172}
					height={68}
					className={styles.logo}
					priority
					quality={60}
				/>
			)}
		</Link>
		<ul className={styles.navigationWrapper}>
			{internalLinksData.map((item) => {
				if (item.path === 'contact') {
					return (
						<div key={item.id} className={styles.navigationMainLinkWrapper}>
							<SwitchCTA
								ctaData={{
									title: item.title,
									link: `/${item?.path}` ?? '',
									externalLink: false,
									buttonType: 'Button tertiary',
								}}
								className={contactButtonClassName}
								color={getColorHeader(color ?? '')}
							/>
						</div>
					)
				}

				if (item.path === 'social-media') return ''

				if (item.path === 'team') {
					return (
						<EmployeeSwitch
							parent={item}
							employeeData={employeesData}
							key={getUniqueId()}
							className={null}
						/>
					)
				}

				return (
					<div key={item.id} className={styles.navigationMainLinkWrapper}>
						<Link
							href={`/${item?.path}`}
							replace
							className={styles.navigationMainLink}
						>
							<li>{item.title}</li>
						</Link>
					</div>
				)
			})}

			<div className={styles.languageSwitch}>
				<LanguageSwitch localizations={localizations} />
			</div>
		</ul>
	</div>
)

export default DesktopNavigation
