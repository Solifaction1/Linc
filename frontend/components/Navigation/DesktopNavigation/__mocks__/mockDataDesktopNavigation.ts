/* eslint-disable max-lines */
import { Navigation } from "@models/navigation.model";

export const navigationMockData: Navigation[] = [{
    audience: [],
    collapsed: false,
    createdAt: 'createdAt',
    id: 1,
    menuAttached: false,
    order: 0,
    path: 'path',
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt',
    items: 'items',
    parent: null,
    related: null
}]
