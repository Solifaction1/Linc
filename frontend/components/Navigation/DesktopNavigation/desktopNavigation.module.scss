@import 'styles/vars';
@import 'styles/utilities';

.navigationWrapper {
    @extend %page-container;
    position: relative;
    display: flex;
    padding: calc(#{$spacing-base} * 5) $page-bleed;
    
    @media (min-width: $break-xlg) {
        padding: calc(#{$spacing-base} * 5) 0 calc(#{$spacing-base} * 5);
    }

    @media (max-width: $break-s) {
        padding: calc(#{$spacing-base} * 5) $page-bleed-mobile;
    }
    
    .navigationWrapper {
        display: flex;
        gap: calc(#{$spacing-base} * 3);
        align-items: center;
        justify-content: flex-end;

        .languageSwitch {
            height: 70%;
        }

        .employeeSwitch {
            height: 75%;
        }

        padding: calc(#{$spacing-base} * 1) 0;

        .navigationMainLinkWrapper {
            position: relative;

            .navigationMainLink {
                display: flex;
                border: none;
                background: none;

                cursor: pointer;

                &:focus-visible {
                    outline: $border-width-md solid $color-white;
                }
            }
        }

        .arrow {
            transition: transform $animation-in ease;

            &.active {
                transform: rotate(-180deg);
            }

            height: 65%; 
        }
    }
}
