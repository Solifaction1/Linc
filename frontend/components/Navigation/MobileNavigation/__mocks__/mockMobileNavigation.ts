/* eslint-disable max-lines */
import { Navigation } from "@models/navigation.model";

export const navigationMockData: Navigation[] = [{
    audience: [],
    collapsed: false,
    createdAt: 'createdAt',
    id: 1,
    menuAttached: false,
    order: 0,
    path: 'path',
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt',
    children: [],
    items: 'items',
    parent: null,
    related: {
        createdAt: '',
        facebook: 'https://www.facebook.com',
        linkedin: 'https://www.linkedin.com',
        instagram: '',
        locale: '',
        id: 1,
        navigationItemId: 2,
        publishedAt: '',
        sitemap_exclude: false,
        updatedAt: '',
        updatedBy: '',
        __contentType: ''
    }
}]
