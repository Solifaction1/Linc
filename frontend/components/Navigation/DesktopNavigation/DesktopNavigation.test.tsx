import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import DesktopNavigation from './DesktopNavigation'
import { navigationMockData } from './__mocks__/mockDataDesktopNavigation'

jest.mock('next/router', () => ({
	useRouter() {
		return {
			locale: 'locale',
		}
	},
}))

describe('DesktopNavigation', () => {
	it('renders a DesktopNavigation', () => {
		const { container, getByText } = render(
			<DesktopNavigation
				internalLinksData={navigationMockData}
				className="className"
				localizations={{ data: undefined }}
				color="dark-green"
				contactButtonClassName="contactButtonClassName"
				employeesData={{ data: [] }}
			/>
		)

		expect(container).toMatchSnapshot()

		expect(
			container.getElementsByClassName('navigationWrapper className')[0]
		).toBeVisible()

		expect(container.querySelector('a[href="/"]')).toBeVisible()
		expect(container.querySelector('img')).toBeVisible()

		expect(container.querySelector('ul')).toBeVisible()
		expect(
			container.getElementsByClassName('navigationWrapper')[0]
		).toBeVisible()
		expect(
			container.getElementsByClassName('navigationMainLinkWrapper')[0]
		).toBeVisible()

		expect(
			container.getElementsByClassName('navigationMainLink')[0]
		).toBeVisible()
		expect(container.querySelector('a[href="/path"]')).toBeVisible()

		expect(container.getElementsByClassName('languageSwitch')[0]).toBeVisible()
		expect(getByText('LOCALE', { selector: 'button' })).toBeVisible()
		expect(
			container.getElementsByClassName('languageLink activeLang')[0]
		).toBeVisible()
		expect(container.querySelectorAll('svg')[0]).toBeVisible()
		expect(container.querySelectorAll('path')[0]).toBeVisible()

		expect(
			container.getElementsByClassName('accordionWrapper')[0]
		).toBeVisible()
		expect(
			container.getElementsByClassName('languageLinkWrapper')[0]
		).toBeVisible()
	})
})
