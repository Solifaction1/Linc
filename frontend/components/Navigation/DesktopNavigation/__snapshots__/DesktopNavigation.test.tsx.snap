// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNavigation renders a DesktopNavigation 1`] = `
<div>
  <div
    class="navigationWrapper className"
  >
    <a
      href="/"
    >
      <img
        alt="logo"
        class="logo"
        data-nimg="1"
        decoding="async"
        height="68"
        src="/images/logo.svg"
        srcset="/images/logo.svg 1x, /images/logo.svg 2x"
        style="color: transparent;"
        width="172"
      />
    </a>
    <ul
      class="navigationWrapper"
    >
      <div
        class="navigationMainLinkWrapper"
      >
        <a
          class="navigationMainLink"
          href="/path"
        >
          <li>
            title
          </li>
        </a>
      </div>
      <div
        class="languageSwitch"
      >
        <div
          class="languageSwitch"
        >
          <button
            class="languageLink activeLang"
            type="button"
          >
            LOCALE
            <svg
              class="false arrow"
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
              />
            </svg>
          </button>
          <div
            class="accordionWrapper"
            style="height: 0px;"
          >
            <div
              class="languageLinkWrapper"
            />
          </div>
        </div>
      </div>
    </ul>
  </div>
</div>
`;
